Rolling upgrades in new format migration broken

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
Server should be running until quorum number of servers are available with new format and not fail. 
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
Fails with 

```
./minio server --config-dir /tmp/minio --address localhost:9004 http://localhost:9001/tmp/c35b1f62-183a-11e8-bdc5-b37298f55263 http://localhost:9002/tmp/c35c16ec-183a-11e8-bd7c-5fd4abdd2ec4 http://localhost:9003/tmp/c35cfb20-183a-11e8-85bf-df5f4f2bfbca http://localhost:9004/tmp/c35e80da-183a-11e8-812e-3ff17a15959b


Trace: 1: cmd/server-main.go:246:cmd.serverMain()
       2: vendor/github.com/minio/cli/app.go:499:cli.HandleAction()
       3: vendor/github.com/minio/cli/command.go:214:cli.Command.Run()
       4: vendor/github.com/minio/cli/app.go:260:cli.(*App).Run()
       5: cmd/main.go:149:cmd.Main()
       6: main.go:71:main.main()
[2018-02-23T01:49:18.512883464Z] [ERROR] Initializing object layer failed (Unsupported XL backend format found [1])
```
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
Avoid failing when we find disks with wrong format versions. 
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
Setup a 4 node setup with GA binary and then kill one server. Start the new master binary on the same backend at this location.

## Context
Large bucket support migration, cloud foundry etc.
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
master and GA binary.

--- a/cmd/auth-rpc-server_test.go
+++ b/cmd/auth-rpc-server_test.go
@@ -52,7 +52,7 @@ func TestLogin(t *testing.T) {
 		{
 			args: LoginRPCArgs{
 				AuthToken: token,
-				Version:   semVersion{2, 0, 0},
+				Version:   semVersion{3, 0, 0},
 			},
 			skewTime:    0,
 			expectedErr: errRPCAPIVersionUnsupported,

--- a/cmd/globals.go
+++ b/cmd/globals.go
@@ -171,8 +171,8 @@ var (
 	// Set to store standard storage class
 	globalStandardStorageClass storageClass
 
-	// RPC version.
-	globalRPCAPIVersion = semVersion{1, 0, 0}
+	// Current RPC version
+	globalRPCAPIVersion = semVersion{2, 0, 0}
 
 	// Add new variable global values here.
 )

